### PR TITLE
feat(feedback): Implement a feedback list page table

### PR DESCRIPTION
**Pull request by ryan953**
Wire up a list of feedbacks and lightly style it.

SCR-20230912-jpdj

Related to https://github.com/getsentry/sentry/issues/55809

### CODE REVIEW

File: static/app/components/feedback/table/feedbackTable.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  location: Location<FeedbackListQueryParams & UrlState>;
+}
+
+const BASE_COLUMNS: GridColumnOrder<string>[] = [
+  {key: 'id', name: 'id'},
+  {key: 'status', name: 'status'},
+  {key: 'contact_email', name: 'contact_email'},
+  {key: 'message', name: 'message'},
+  {key: 'timestamp', name: 'timestamp'},
+];
```
Comment (billyvg) [inline static/app/components/feedback/table/feedbackTable.tsx:39]: TODO: add replay column?

File: static/app/components/feedback/table/feedbackTable.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    <Link
+      to={{
+        pathname: normalizeUrl(`/organizations/${organization.slug}/feedback/${value}/`),
+        query: {referrer: 'feedback_list_page'},
+      }}
+      onClick={() => {
+        trackAnalytics('feedback_list.details_link.click', {organization});
+      }}
+    >
+      {value}
```
Comment (billyvg) [inline static/app/components/feedback/table/feedbackTable.tsx:118]: Thoughts on using [logaf scale](https://blog.danlew.net/2020/04/15/the-logaf-scale/)?

[m]edium: do we have a "shorten id" util function we can use? Not useful to see the full uuid

File: static/app/views/feedback/list.tsx [hunk 3]
```diff
@@ -1,17 +1,28 @@
...
 import SentryDocumentTitle from 'sentry/components/sentryDocumentTitle';
 import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {FeedbackListQueryParams} from 'sentry/utils/feedback/types';
 import useOrganization from 'sentry/utils/useOrganization';
 
-export default function List() {
+interface Props extends RouteComponentProps<{}, {}, FeedbackListQueryParams> {}
+
+export default function List({location}: Props) {
```
Comment (billyvg) [inline static/app/views/feedback/list.tsx:22]: [l]ow: generally like to see component names w/ more context in the name (e.g. FeedbackList) as it's a bit more helpful when debugging w/ react dev tools, but we don't have a rule for this (and also hard to have a lintable rule) so low logaf.

File: static/app/views/feedback/list.tsx [hunk 4]
```diff
@@ -31,18 +42,43 @@ export default function List() {
...
+                data={data ?? []}
+                isError={isError}
+                isLoading={isLoading}
+                location={location}
+              />
+            </LayoutGap>
+            <PaginationNoMargin
+              pageLinks={
+                pageLinks ||
+                '<https://sentry.sentry.io/api/0/organizations/sentry/replays/?field=activity&field=browser&field=count_dead_clicks&field=count_errors&field=count_rage_clicks&field=duration&field=finished_at&field=id&field=is_archived&field=os&field=project_id&field=started_at&field=urls&field=user&per_page=50&project=11276&query=&sort=-started_at&statsPeriod=24h&cursor=0:0:1>; rel="previous"; results="false"; cursor="0:0:1", <https://sentry.sentry.io/api/0/organizations/sentry/replays/?field=activity&field=browser&field=count_dead_clicks&field=count_errors&field=count_rage_clicks&field=duration&field=finished_at&field=id&field=is_archived&field=os&field=project_id&field=started_at&field=urls&field=user&per_page=50&project=11276&query=&sort=-started_at&statsPeriod=24h&cursor=0:50:0>; rel="next"; results="true"; cursor="0:50:0"'
```
Comment (billyvg) [inline static/app/views/feedback/list.tsx:61]: Should this be removed? or maybe mark with `TODO` if we need to keep it temporarily

**Review by billyvg — APPROVED** [review]

File: static/app/components/feedback/table/feedbackTable.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/components/feedback/table/feedbackTable.tsx:118]: [logaf scale](https://blog.danlew.net/2020/04/15/the-logaf-scale/) 👍

File: static/app/views/feedback/list.tsx (hunk 4 above)
Comment (ryan953) [reply static/app/views/feedback/list.tsx]: ya, removed

File: static/app/views/feedback/list.tsx (hunk 3 above)
Comment (ryan953) [reply static/app/views/feedback/list.tsx:22]: ya, this name sucks. I can pick a unique one to represent it better

File: static/app/components/feedback/table/feedbackTable.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/components/feedback/table/feedbackTable.tsx:118]: `<ShortId>`

File: static/app/components/feedback/table/feedbackTable.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/components/feedback/table/feedbackTable.tsx:39]: done, lol!

**Pull request merged**